aggregator/src/aggregator.rs

```diff
@@ -3526,80 +3536,88 @@ mod tests {
         )
     }
 
-    pub(super) fn create_report(task: &AggregatorTask, report_timestamp: Time) -> Report {
-        create_report_custom(task, report_timestamp, random(), task.current_hpke_key())
+    pub(super) fn create_report(
+        task: &AggregatorTask,
+        hpke_keypair: &HpkeKeypair,
+        report_timestamp: Time,
+    ) -> Report {
+        create_report_custom(task, report_timestamp, random(), hpke_keypair)
     }
 
-    async fn setup_upload_test(
-        cfg: Config,
-    ) -> (
-        Prio3Count,
-        Aggregator<MockClock>,
-        MockClock,
-        Task,
-        Arc<Datastore<MockClock>>,
-        EphemeralDatastore,
-    ) {
-        setup_upload_test_with_runtime(TestRuntime::default(), cfg).await
+    struct UploadTest {
+        vdaf: Prio3Count,
+        aggregator: Aggregator<MockClock>,
+        clock: MockClock,
+        task: Task,
+        datastore: Arc<Datastore<MockClock>>,
+        ephemeral_datastore: EphemeralDatastore,
+        hpke_keypair: HpkeKeypair,
     }
```

Comment:
This file has always been too big. If we're touching a lot of the test code anyway, then it could be wise to move `struct UploadTest` and supporting code to some dedicated test module like `aggregator/src/aggregator/aggregator_test.rs`. I think that makes using `struct UploadTest` from other test modules a little more graceful, too.